controllers/capability/statefulset.go

```diff
@@ -0,0 +1,331 @@
+package capability
+
+import (
+	"encoding/json"
+	"fmt"
+	"hash/fnv"
+	"strconv"
+
+	"github.com/Dynatrace/dynatrace-operator/api/v1alpha1"
+	"github.com/Dynatrace/dynatrace-operator/controllers/customproperties"
+	"github.com/Dynatrace/dynatrace-operator/controllers/dtpullsecret"
+	"github.com/Dynatrace/dynatrace-operator/controllers/utils"
+	"github.com/pkg/errors"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+const (
+	serviceAccountPrefix = "dynatrace-"
+
+	kubernetesArch     = "kubernetes.io/arch"
+	kubernetesOS       = "kubernetes.io/os"
+	kubernetesBetaArch = "beta.kubernetes.io/arch"
+	kubernetesBetaOS   = "beta.kubernetes.io/os"
+
+	amd64 = "amd64"
+	arm64 = "arm64"
+	linux = "linux"
+
+	annotationTemplateHash    = "internal.operator.dynatrace.com/template-hash"
+	annotationImageHash       = "internal.operator.dynatrace.com/image-hash"
+	annotationImageVersion    = "internal.operator.dynatrace.com/image-version"
+	annotationCustomPropsHash = "internal.operator.dynatrace.com/custom-properties-hash"
+
+	DTCapabilities    = "DT_CAPABILITIES"
+	DTIdSeedNamespace = "DT_ID_SEED_NAMESPACE"
+	DTIdSeedClusterId = "DT_ID_SEED_K8S_CLUSTER_ID"
+
+	DTCapabilitiesArg = "--enable=$(DT_CAPABILITIES)"
+
+	ProxyArg = `PROXY="${ACTIVE_GATE_PROXY}"`
+	ProxyEnv = "ACTIVE_GATE_PROXY"
+	ProxyKey = "ProxyKey"
+
+	keyModule = "module"
+)
+
+type StatefulSetEvent func(sts *appsv1.StatefulSet)
+
+type statefulSetProperties struct {
+	*v1alpha1.DynaKube
+	*v1alpha1.CapabilityProperties
+	customPropertiesHash  string
+	kubeSystemUID         types.UID
+	module                string
+	capabilityName        string
+	serviceAccountOwner   string
+	onAfterCreateListener []StatefulSetEvent
+}
+
+func NewStatefulSetProperties(instance *v1alpha1.DynaKube, capabilityProperties *v1alpha1.CapabilityProperties,
+	kubeSystemUID types.UID, customPropertiesHash string, module string, capabilityName string, serviceAccountOwner string) *statefulSetProperties {
+	if serviceAccountOwner == "" {
```

Comment:
serviceAccountOwner basically exists to support kubernetes monitoring, since it uses `kubemon` as feature name and "kubernetes-monitoring" as a part of the service-accounts name. For future capabilities, I thought it would be better to use the feature name for the latter part, which is why `feature` (`module` before the latest changes) overwrites `serviceAccountName` if `serviceAccountName` is empty.

Reply:
OK, got it.